src/data/list/nthd.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2022 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+
+import data.list.defs
+
+/-!
+# Accessing lists with default values
+
+-/
+
+namespace list
+
+section nthd
+
+variables {α : Type*} (l : list α) (x : α) (xs : list α) (d : α) (n : ℕ)
+
+/-- "default" `nth` function: returns `d` instead of `none` in the case
+  that the index is out of bounds. -/
+def nthd : Π (l : list α) (n : ℕ), α
```

Comment:
Sorry, I commented on the downstream PR by mistake. Why not use use `@inth _ ⟨d⟩ l n`?

Reply:
Does somethiing like `@inth_eq_nth_le _ _ _ ⟨d⟩ _` work? Or `by haveI : inhabited _ := _; exact inth_eq_nth_le`?

Reply:
What if we redefine `inth` instead?

Reply:
That seems very reasonable to me